File: examples/python/langchain_simple.py
from langchain_hyperspace import HyperspaceVectorStore, YARLabsEmbeddings
from langchain_core.documents import Document

def main():
    # 1. Setup YARLabs Local Embeddings
    print("Loading YARLabs Local Embedding Model (0.5B)...")
    embeddings = YARLabsEmbeddings()
    
    # 2. Initialize Store
    # This will automatically create the collection if it doesn't exist
    vectorstore = HyperspaceVectorStore(
        host="localhost",
        port=50051,
        collection_name="langchain_python_demo",
        embedding_function=embeddings,
        api_key="I_LOVE_HYPERSPACEDB"
    )

    # 3. Add Documents
    print("Adding documents...")
    documents = [
        Document(
            page_content="HyperspaceDB is a multi-geometry vector database.",
            metadata={"category": "tech", "level": "advanced"}
        ),
        Document(
            page_content="LangChain is a framework for developing AI applications.",
            metadata={"category": "ai", "level": "beginner"}
        )
    ]
    ids = vectorstore.add_documents(documents)
    print(f"Added {len(ids)} documents with IDs: {ids}")

    # 4. Simple Similarity Search
    print("Searching for 'vector database'...")
    results = vectorstore.similarity_search("vector database", k=1)
    if results:
        print(f"Top result: {results[0].page_content}")

    # 5. Search with Scores
    print("Searching for 'AI' with scores...")
    results_with_score = vectorstore.similarity_search_with_score("AI", k=1)
    for doc, score in results_with_score:
        print(f"Result: {doc.page_content} (Distance: {score})")

    # 6. Delete Documents
    # delete from the first added ID
    to_delete = ids[0:1]
    print(f"Deleting documents with IDs: {to_delete}...")
    vectorstore.delete(ids=to_delete)
    
    # Cleanup channel
    del vectorstore

if __name__ == "__main__":
    main()

File: examples/ts/langchain_simple.ts
import { HyperspaceStore, YARLabsEmbeddings } from "../../integrations/langchain-js/src/index";
import { HyperspaceClient } from "../../sdks/ts/src/client";
import { Document } from "@langchain/core/documents";

async function main() {
    const client = new HyperspaceClient("localhost:50051", "I_LOVE_HYPERSPACEDB");
    
    // 2. Initialize YARLabs Local Embeddings (requires @xenova/transformers)
    const embeddings = new YARLabsEmbeddings();
    
    // 1. Initialize Store
    const store = new HyperspaceStore(embeddings, {
        client,
        collectionName: "langchain_js_lorentz_demo",
        // The YARLabs 0.5B model with targetDim=64 produces 65-dimensional Lorentz vectors (1 + 64)
        dimension: 65,
        metric: "lorentz",
    });

    // 2. Add Documents
    console.log("Adding documents...");
    await store.addDocuments([
        new Document({ 
            pageContent: "HyperspaceDB is a multi-geometry vector database.",
            metadata: { category: "tech", level: "advanced" } 
        }),
        new Document({ 
            pageContent: "LangChain is a framework for developing AI applications.",
            metadata: { category: "ai", level: "beginner" } 
        })
    ]);

    // 3. Similarity Search
    console.log("Searching for 'vector database'...");
    const results = await store.similaritySearch("vector database", 1);
    console.log("Top result:", results[0].pageContent);

    // 4. Search with Metadata Filter
    console.log("Searching for 'AI' with category=ai filter...");
    const filteredResults = await store.similaritySearch("AI", 1, { category: "ai" });
    console.log("Filtered result:", filteredResults[0].pageContent);

    // 5. Delete by ID
    const demoId = "12345";
    await store.addDocuments([new Document({ pageContent: "Temp doc", metadata: { id: demoId } })], { ids: [demoId] });
    console.log("Deleting document...");
    await store.delete({ ids: [demoId] });
    
    client.close();
}

main().catch(console.error);
